File: web/src/model/victron_controller/dashboard/from_0_2_0_sensors.ts
// @ts-nocheck
import {Sensors as v0_2_0_Sensors} from './v0_2_0/Sensors'
import {Sensors as dashboard_Sensors} from './Sensors'

export function convert__sensors__from__0_2_0(from: v0_2_0_Sensors): dashboard_Sensors {
    return new dashboard_Sensors (
        JSON.parse(JSON.stringify(from.battery_soc)),
        JSON.parse(JSON.stringify(from.battery_soh)),
        JSON.parse(JSON.stringify(from.battery_installed_capacity)),
        JSON.parse(JSON.stringify(from.battery_dc_power)),
        JSON.parse(JSON.stringify(from.mppt_power_0)),
        JSON.parse(JSON.stringify(from.mppt_power_1)),
        JSON.parse(JSON.stringify(from.soltaro_power)),
        JSON.parse(JSON.stringify(from.power_consumption)),
        JSON.parse(JSON.stringify(from.grid_power)),
        JSON.parse(JSON.stringify(from.grid_voltage)),
        JSON.parse(JSON.stringify(from.grid_current)),
        JSON.parse(JSON.stringify(from.consumption_current)),
        JSON.parse(JSON.stringify(from.offgrid_power)),
        JSON.parse(JSON.stringify(from.offgrid_current)),
        JSON.parse(JSON.stringify(from.vebus_input_current)),
        JSON.parse(JSON.stringify(from.evcharger_ac_power)),
        JSON.parse(JSON.stringify(from.evcharger_ac_current)),
        JSON.parse(JSON.stringify(from.ess_state)),
        JSON.parse(JSON.stringify(from.outdoor_temperature)),
        JSON.parse(JSON.stringify(from.session_kwh)),
        JSON.parse(JSON.stringify(from.ev_soc)),
        JSON.parse(JSON.stringify(from.ev_charge_target))
    )
}

File: crates/dashboard-model/src/victron_controller/dashboard/from_0_2_0_sensors.rs


pub fn convert__sensors__from__0_2_0(from: &crate::victron_controller::dashboard::v0_2_0::sensors::Sensors) -> crate::victron_controller::dashboard::sensors::Sensors {
    crate::victron_controller::dashboard::sensors::Sensors {
        battery_soc: serde_json::from_value(serde_json::to_value(&from.battery_soc).unwrap()).unwrap(),
        battery_soh: serde_json::from_value(serde_json::to_value(&from.battery_soh).unwrap()).unwrap(),
        battery_installed_capacity: serde_json::from_value(serde_json::to_value(&from.battery_installed_capacity).unwrap()).unwrap(),
        battery_dc_power: serde_json::from_value(serde_json::to_value(&from.battery_dc_power).unwrap()).unwrap(),
        mppt_power_0: serde_json::from_value(serde_json::to_value(&from.mppt_power_0).unwrap()).unwrap(),
        mppt_power_1: serde_json::from_value(serde_json::to_value(&from.mppt_power_1).unwrap()).unwrap(),
        soltaro_power: serde_json::from_value(serde_json::to_value(&from.soltaro_power).unwrap()).unwrap(),
        power_consumption: serde_json::from_value(serde_json::to_value(&from.power_consumption).unwrap()).unwrap(),
        grid_power: serde_json::from_value(serde_json::to_value(&from.grid_power).unwrap()).unwrap(),
        grid_voltage: serde_json::from_value(serde_json::to_value(&from.grid_voltage).unwrap()).unwrap(),
        grid_current: serde_json::from_value(serde_json::to_value(&from.grid_current).unwrap()).unwrap(),
        consumption_current: serde_json::from_value(serde_json::to_value(&from.consumption_current).unwrap()).unwrap(),
        offgrid_power: serde_json::from_value(serde_json::to_value(&from.offgrid_power).unwrap()).unwrap(),
        offgrid_current: serde_json::from_value(serde_json::to_value(&from.offgrid_current).unwrap()).unwrap(),
        vebus_input_current: serde_json::from_value(serde_json::to_value(&from.vebus_input_current).unwrap()).unwrap(),
        evcharger_ac_power: serde_json::from_value(serde_json::to_value(&from.evcharger_ac_power).unwrap()).unwrap(),
        evcharger_ac_current: serde_json::from_value(serde_json::to_value(&from.evcharger_ac_current).unwrap()).unwrap(),
        ess_state: serde_json::from_value(serde_json::to_value(&from.ess_state).unwrap()).unwrap(),
        outdoor_temperature: serde_json::from_value(serde_json::to_value(&from.outdoor_temperature).unwrap()).unwrap(),
        session_kwh: serde_json::from_value(serde_json::to_value(&from.session_kwh).unwrap()).unwrap(),
        ev_soc: serde_json::from_value(serde_json::to_value(&from.ev_soc).unwrap()).unwrap(),
        ev_charge_target: serde_json::from_value(serde_json::to_value(&from.ev_charge_target).unwrap()).unwrap(),
    }
}